### PR TITLE
Repo setup

### LINKED ISSUE #23: create repo and skelton
title

### PR DESCRIPTION
**Pull request by celery6**
closes #23

### CODE REVIEW

**Review by Armaan-Sengupta — DISMISSED** [review]
I just added command to update submodule in readme, we should also link an STM32 first time setup guide (christine you made a tutorial on this right?). And some pics of the ST Link would be nice, but yea LGTM

**Comment by celery6** [comment]
> I just added command to update submodule in readme, we should also link an STM32 first time setup guide (christine you made a tutorial on this right?). And some pics of the ST Link would be nice, but yea LGTM

imo readmes are not tutorials, hence i didn't put any detailed tutorials of how to do the things i referenced. The "how to" information is elsewhere; the readme just says the "what"

File: README.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
-# cansw_processor_stm
+# cansw_processor_canards
 
-Software repo for the 2025 canards processor board.
+ Firmware for the 2024-2025 processor board project, performing state estimation and control for the canards system.
+ Project documentation can be found in the team Google Drive.
+
+## Developer Setup
+1. Clone repo locally
```
Comment (JasonBrave) [inline README.md:7]:
```suggestion
git clone --recurse-submodules git@github.com:waterloo-rocketry/cansw_processor_canards.git
```

Comment (JasonBrave) [reply README.md:7]: So you don't have to get submodules with a separate command

File: README.md [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+   ```bash
+   git submodule update --init --recursive
+   ```
+4. Open project in STM32CubeIDE (recommended IDE)
+5. Use ST-Link to flash code to processor board
+
+## Project Structure
+- `STM32Cube/src/drivers/`: custom peripheral driver modules
+- `STM32Cube/src/application/`: high-level application logic modules
+- `STM32Cube/src/lib/`: third-party libraries
```
Comment (JasonBrave) [inline README.md:18]: IIRC when we had the kickoff meeting we were planning to have a different folder name for 3rd arty library flder :face:

**Review by JasonBrave — CHANGES_REQUESTED** [review]

**Comment by JasonBrave** [comment]
Otherwise LGTM

**Review by JasonBrave — CHANGES_REQUESTED** [review]
1. typo
image
2. Keep everything lower case
image
3. What's the point of SD card driver, the CubeMX connects FATFS to SDMMC for you
4. Delete .DS_Store and add it to gitignore
5. More typo
image
5. canlib should be on comp-2024 git tag

File: .DS_Store
Comment (martendo) [inline .DS_Store:1]: delete

File: Core/.DS_Store
Comment (martendo) [inline Core/.DS_Store:1]: delete

File: .gitignore [hunk 3]
```diff
@@ -1,7 +1,12 @@
...
+Debug/*
+Release/*
 **/.metadata
 **/.settings
 **/.vscode
 *.orig
+*.elf
+*.o
+
+.settings
```
Comment (martendo) [inline .gitignore:11]: `**/foo` is equivalent to `foo` (https://git-scm.com/docs/gitignore#_pattern_format) so having `.settings` twice is redundant
```suggestion
.metadata
.settings
.vscode
*.orig
*.elf
*.o
```

File: .gitignore
Comment (martendo) [inline .gitignore:1]: No newline at end of file :(

**Review by martendo — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (celery6) [reply .gitignore]: good catch, this was copypasted from last year rip

File: .gitignore
Comment (celery6) [reply .gitignore:1]: also good catch. that is very sad :(

**Review by martendo — APPROVED** [review]

**Review by JasonBrave — APPROVED** [review]
LGTM

**Pull request merged**